inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -368,19 +369,32 @@ DeviceInformation MultiDeviceInferencePlugin::SelectDevice(const std::vector<Dev
             if (gpuFullDeviceName.find("iGPU") != std::string::npos) {
                 iGPU.push_back(item);
             } else if (gpuFullDeviceName.find("dGPU") != std::string::npos) {
-                dGPU.push_back(item);
+                // Fixme:  FULL_DEVICE_NAME : Intel(R) Graphics [0x4f87] (dGPU) for DG2 now
+                if (gpuFullDeviceName.find("0x4f87") != std::string::npos) {
+                    dGPU_DG2.push_back(item);
+                } else if (gpuFullDeviceName.find("Intel(R) Iris(R) Xe MAX Graphics") != std::string::npos) {
+                    dGPU_DG1.push_back(item);
+                }
             }
             continue;
         }
     }
 
-    if (CPU.empty() && dGPU.empty() && iGPU.empty() && MYRIAD.empty() && VPUX.empty()) {
+    if (CPU.empty() && dGPU_DG2.empty() && dGPU_DG1.empty() && iGPU.empty() && MYRIAD.empty() && VPUX.empty()) {
         IE_THROW(NotFound) << "No available device found";
     }
 
-    // Priority of selecting device: dGPU > VPUX > iGPU > MYRIAD > CPU
-    if (!dGPU.empty()) {
-        for (auto&& item : dGPU) {
+    // Priority of selecting device: dGPU DG2 > dGPU DG1 > VPUX > iGPU > MYRIAD > CPU
```

Comment:
this is why we need a gigaflops metric supported by the device, to avoid adding priority for each and every device manually